101-cloud-shell-vnet/azuredeploy.json

```diff
@@ -81,12 +81,15 @@
         "relayNamespaceRef": "[resourceId('Microsoft.Relay/namespaces/', parameters('relayNamespaceName'))]",
         "relaySubnetRef": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('existingVNETName'), parameters('relaySubnetName'))]",
         "contributorRoleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')]",
-        "networkRoleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', '4d97b98b-1d4f-4787-a291-c67834d212e7')]"
+        "networkRoleDefinitionId": "[resourceId('Microsoft.Authorization/roleDefinitions', '4d97b98b-1d4f-4787-a291-c67834d212e7')]",
+        "privateDnsZoneName": "privatelink.servicebus.windows.net",
```

Comment:
this cannot be hardcoded - you can pass it in as a param

Reply:
Line 85 - the privateDnsZoneName

Reply:
@bmoore-msft: If I make it a parameter and users plug in a different value, this wouldn't work anymore. Is there a way to make the paramter ready only?

Reply:
If it can only be one possible value, why is it required that the user supply it?

Reply:
User is not required to supply it. That is why I had it declared as a variable instead of parameter.

Reply:
Does it have to be provided in the PUT request?

Reply:
yes, it's required for PUT request

Reply:
That's the part that doesn't make sense - it's required on the PUT but it cannot be changed.  If it cannot be changed, why is it required on the PUT?

Reply:
We need to create a private DNS zone for our vnet. the name needs to be privatelink.servicebus.windows.net as we need to add a dns record for relay.privatelink.servicebus.winodws.net -> private ip address. If we make it a parameter and user changes it to something else , it won't be translated correctly. Does this make sense?

Reply:
Yes, it sounds like that property value can only be the exact string you've specified.  If it cannot be changed, why does it have to be specified in the template?

Reply:
@bmoore-msft : I don't understand. If we don't specify it in template how can i create a private dns zone resource with that name?

Reply:
I think the part I'm struggling with is why we have an API in Azure that requires a name that contains an Azure endpoint (and it cannot be changed).  That's a strange/error prone pattern and  will prevent that template from being used across clouds.

Reply:
@bmoore-msft : Private DNS zones can be created for different resources and would have a different value for different scenarios. For out particular scenario, it needs to be privatelink.servicebus.windows.net but it can be different depending on what resources you have.